front/src/applications/operationalStudies/components/MicroMacroSwitch.tsx

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+
+import cx from 'classnames';
+
+type MicroMacroSwitchProps = {
+  isMacro: boolean;
+  onClickButton: (isMacro: boolean) => void;
+  t: (key: string) => string;
+};
+
+const MicroMacroSwitch = ({ isMacro, onClickButton, t }: MicroMacroSwitchProps) => (
```

Comment:
You can get the translation tool directly here:
```suggestion
const MicroMacroSwitch = ({ isMacro, onClickButton }: MicroMacroSwitchProps) => {
  const { t } = useTranslation('operationalStudies/scenario');
  return (
```

Reply:
Done